src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs

```diff
@@ -7607,6 +7607,368 @@ void Method()
 
     void Goo(ref bool b) { }
     void Goo(int i) { }
+}");
+        }
+
+        [WorkItem(26993, "https://github.com/dotnet/roslyn/issues/26993")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateFieldInExpressionBodiedGetter()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    public int Y
+    {
+        get => [|y|];
+    }
+}",
+@"class Program
+{
+    private int y;
+
+    public int Y
+    {
+        get => y;
+    }
+}");
+        }
+
+        [WorkItem(26993, "https://github.com/dotnet/roslyn/issues/26993")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateFieldInExpressionBodiedGetterWithDifferentAccessibility()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    public int Y
+    {
+        protected get => [|y|];
+        set => throw new System.NotImplementedException();
+    }
+}",
+@"class Program
+{
+    private int y;
+
+    public int Y
+    {
+        protected get => y;
+        set => throw new System.NotImplementedException();
+    }
+}");
+        }
+
+        [WorkItem(26993, "https://github.com/dotnet/roslyn/issues/26993")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateReadonlyFieldInExpressionBodiedGetter()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    public int Y
+    {
+        get => [|y|];
+    }
+}",
+@"class Program
+{
+    private readonly int y;
+
+    public int Y
+    {
+        get => y;
+    }
+}",
+index: 1);
```

Comment:
>1 [](start = 7, length = 1)

I know other tests already do this, but it would be nicer to define a couple of constants for these indices. It's not immediately clear what these mean.